improvement: CLDSRV-61 refactor and test versioning helpers

jonathan-gramain opened the pull request:
- add unit tests for processVersioningState helper

- remove the callback argument, instead, return the list of parameters
  as an object, it simplifies and enhances testability

- add unit tests for preprocessingVersioningDelete helper

jonathan-gramain committed:
improvement: CLDSRV-61 refactor and test processVersioningState

- add unit tests for processVersioningState() helper

- remove the callback argument, instead, return the list of parameters
  as an object, it simplifies and enhances testability

jonathan-gramain committed:
improvement: CLDSRV-61 test preprocessingVersioningDelete

Add unit tests for preprocessingVersioningDelete helper

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-61 contains:


* `7.10.2`

* `7.4.11`

* `8.3.2`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.2`

* `7.4.11`

* `8.2.18`

* `8.3.2`


Please check the `Fix Version/s` of CLDSRV-61, or the target
branch of this pull request.

jonathan-gramain commented:
ping

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CLDSRV-61-refactorAndTestVersioningHelpers` into
`development/7.4`
* [w/7.10/improvement/CLDSRV-61-refactorAndTestVersioningHelpers](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/improvement/CLDSRV-61-refactorAndTestVersioningHelpers) will be merged into `development/7.10` (pull request #4072)
* [w/8.2/improvement/CLDSRV-61-refactorAndTestVersioningHelpers](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/improvement/CLDSRV-61-refactorAndTestVersioningHelpers) will be merged into `development/8.2` (pull request #4073)
* [w/8.3/improvement/CLDSRV-61-refactorAndTestVersioningHelpers](https://github.com/scality/cloudserver/compare/development/8.3...w/8.3/improvement/CLDSRV-61-refactorAndTestVersioningHelpers) will be merged into `development/8.3` (pull request #4074)



The following branches will **NOT** be impacted:

* `development/6.4`
* `stabilization/7.10.1`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

watsaqat commented on lib/api/apiUtils/object/versioning.js:
```diff
@@ -212,35 +212,29 @@ function getMasterState(objMD) {
  */
 function versioningPreprocessing(bucketName, bucketMD, objectKey, objMD,
```
same feedback as above

jonathan-gramain replied:
This function has jsdoc already, not sure if you are referring to something more specific

watsaqat commented on lib/api/apiUtils/object/versioning.js:
```diff
@@ -128,7 +128,7 @@ function _deleteNullVersionMD(bucketName, objKey, options, mst, log, cb) {
 }
 
-function processVersioningState(mst, vstat, cb) {
+function processVersioningState(mst, vstat) {
```
jsDoc would be helpful; not obvious what the params should be

jonathan-gramain committed:
CLDSRV-61 add JSDoc

Add JSDoc for processVersioningState() and getMasterState() versioning
helper functions

jonathan-gramain commented:
ping

jonathan-gramain commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/647d8135b514a794e297d3d7081404863d043f53) did not succeed in branch w/8.3/improvement/CLDSRV-61-refactorAndTestVersioningHelpers.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`



The following branches will **NOT** be impacted:

* `development/6.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`



The following branches have **NOT** changed:

* `development/6.4`



Please check the status of the associated issue CLDSRV-61.

Goodbye jonathan-gramain.
